node/src/crypto/asymmetric_key_ext.rs

```diff
@@ -93,13 +93,13 @@ impl AsymmetricKeyExt for SecretKey {
     fn generate_ed25519() -> Result<Self, Error> {
         let mut bytes = [0u8; Self::ED25519_LENGTH];
         getrandom::getrandom(&mut bytes[..]).expect("RNG failure!");
-        Ok(SecretKey::ed25519(bytes))
+        Ok(SecretKey::ed25519_from_bytes(bytes).unwrap())
     }
 
     fn generate_secp256k1() -> Result<Self, Error> {
         let mut bytes = [0u8; Self::SECP256K1_LENGTH];
         getrandom::getrandom(&mut bytes[..]).expect("RNG failure!");
-        Ok(SecretKey::secp256k1(bytes))
+        Ok(SecretKey::secp256k1_from_bytes(bytes).unwrap())
```

Comment:
Why the `unwrap`?  We're returning a `Result<Self, Error>`...

Reply:
`unwrap` removed